CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,157 @@
+# How to Contribute to Jaeger
+
+We'd love your help!
+
+Jaeger is [Apache 2.0 licensed](./LICENSE) and accepts contributions via GitHub
+pull requests. This document outlines some of the conventions on development
+workflow, commit message formatting, contact points and other resources to make
+it easier to get your contribution accepted.
+
+We gratefully welcome improvements to documentation as well as to code.
+
+Table of Contents:
+
+* [Making a Change](#making-a-change)
+* [License](#license)
+* [Certificate of Origin - Sign your work](#certificate-of-origin---sign-your-work)
+* [Branches](#branches)
+
+## Making a Change
+
+**Before making any significant changes, please open an issue**. Each issue
+should describe the following:
+
+* Requirement - what kind of business use case are you trying to solve?
+* Problem - what in Jaeger blocks you from solving the requirement?
+* Proposal - what do you suggest to solve the problem or improve the existing
+  situation?
+* Any open questions to address
+
+Discussing your proposed changes ahead of time will make the contribution
+process smooth for everyone. Once the approach is agreed upon, make your changes
+and open a pull request (PR). Each PR should describe:
+
+* Which problem it is solving. Normally it should be simply a reference to the
+  corresponding issue, e.g. `Resolves #123`.
+* What changes are made to achieve that.
+
+Your pull request is most likely to be accepted if **each commit**:
+
+* Has a [good commit message][good-commit-msg]. In summary:
+  * Separate subject from body with a blank line
+  * Limit the subject line to 50 characters
+  * Capitalize the subject line
+  * Do not end the subject line with a period
+  * Use the imperative mood in the subject line
+  * Wrap the body at 72 characters
+  * Use the body to explain _what_ and _why_ instead of _how_
+* Has been signed by the author ([see below](#certificate-of-origin---sign-your-work)).
+
+## License
+
+By contributing your code, you agree to license your contribution under the
+terms of the [Apache License](./LICENSE).
+
+If you are adding a new file it should have a header like below. In some
+languages, e.g. Python, you may need to change the comments to start with `#`.
+The easiest way is to copy the header from one of the existing source files and
+make sure the year is current and the copyright says "The Jaeger Authors".
+
+```
+// Copyright (c) 2018 The Jaeger Authors.
```

Comment:
Probably not something to fix here, but at the source. The year probably needs to be updated ;)